### PR TITLE
Build C stubs for Xen too

### PR DESCRIPTION
**Pull request by talex5**

### CODE REVIEW

**Comment by avsm** [comment]
Any progress on reviewing/merging this patch?

File: src/cipher_block.ml [hunk 1]
```diff
@@ -350,7 +350,7 @@ module DES = struct
     let e_of_secret cs = of_secret ~direction:D3DES.en0 cs
     and d_of_secret cs = of_secret ~direction:D3DES.de1 cs
 
-    let encrypt_block ~key src dst =
+    let encrypt_block ~(key:ekey) src dst =
```
Comment (hannesm) [inline src/cipher_block.ml]: is this change intentional?

**Comment by avsm** [comment]
Ping...any progress on reviewing/merging this patch?

**Comment by pqwy** [comment]
@talex5 -- Sorry for the delay, I've been MIA for various reasons in the past month.

Got around to actually testing the setup and it works wonderfully.

Not sure if the approach of modifying the build system of any old library you'd want to run on xen, to build the C bits twice, will be tenable further on, but it does its job as a quick fix for the TLS stack flawlessly.

Thanks!

**Comment by talex5** [comment]
Agreed - the long term fix is support for cross-compiling in opam, but it seems that's a long way off.

**Comment by pqwy** [comment]
Agreed on both counts.

**Comment by avsm** [comment]
Also agreed that cross-compiling is the future, but this is an extremely useful checkpoint.

**Comment by avsm** [comment]
Could we have a release with this in it please?
